power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -364,6 +373,17 @@ class Deserializer {
     using DataByteMeta = std::vector<std::vector<ComponentByteMeta>>;
     using AttributeByteMeta = std::vector<std::pair<std::string_view, std::vector<std::string_view>>>;
 
+    struct BufferView {
+        WritableDataset::Buffer const* buffer{nullptr};
+        Idx idx{0};
+        std::unordered_map<MetaAttribute const*, AttributeBuffer<void> const*> const* attribute_order{nullptr};
```

Comment:
I do not think we need a `map` or `unordered_map`. A `vector` should be enough. It's a index lookup.

Reply:
i initially had done it with `std::vector` as well. however, using `std::unordered_map` means it can be used to speed up map-like as well as array-like representations.

Using the `std::vector` version would only speed up the array-like representation, while the map-like representation remains linear in complexity.

Since, on average, look-up is `O(1)` when using `std::unordered_map`, i don't see any disadvantages with using `std::unordered_map` compared to `std::vector`. In addition, the creation of the look-up table is much simpler this way.

Reply:
What do you mean by map-like elements? We are talking about the condense format right?

The non-condense format with named attribute, you need a name lookup anyway. That's a separate routine with its own data structure.

In general, `vector` is much faster than `std::unordered_map`. The `O(1)` does not take the constant into account.

Reply:
i've rewritten it to use vector. it does make the logic a lot more convoluted, though. the i've had to add a lot of asserts to even convince myself that what i need is indeed valid in that location. fortunately, asserts are free.